app/models/coupon.rb

```diff
@@ -0,0 +1,10 @@
+class Coupon < ApplicationRecord
+
+  belongs_to :user
+
+  validates :code, :expiry_date, presence: true
+  validates :code, uniqueness: { case_sensitive: false }
+
+  enum status: [:active, :expired]
```

Comment:
what's the advantage of storing the status?

Reply:
Yes sir, it can be avoided, just we have to create scopes and methods for the same to compare using expiry_date.

Reply:
Sir, status removed.